// Copyright 2020 Fuji-Iot authors. All rights reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package fuji_iot.proto;

service FujiACControllerService {
  rpc GetStatus(StatusRequest) returns (StatusResponse) {}
  // Set new state and return status.
  rpc Update(UpdateRequest) returns (StatusResponse) {}
}

enum Mode {    
    MODE_UNKNOWN = 0;
    MODE_FAN = 1;
    MODE_DRY = 2;
    MODE_COOL = 3;
    MODE_HEAT = 4;
    MODE_AUTO = 5;
    MODE_OFF = 6;
}

enum Fan {    
    FAN_UNKNOWN = 0;
    FAN_AUTO = 1;
    FAN_LOW = 2;
    FAN_MEDIUM = 3;
    FAN_HIGH = 4;
    FAN_MAX = 5;
}

// Represents AC unit state.
message ACUnitState {
  Mode mode = 1;
  Fan fan = 2;  
  int32 setpoint_temperature = 3;  
}

message StatusRequest{}

message StatusResponse{
  ACUnitState state = 1;
}

// updates AC Unit state. Fields that are left not-set will maintain the current value.
// Send empty request to get a simple status response without altering AC state.
message UpdateRequest{
  ACUnitState new_state = 1;
}
